mirrord-layer/src/lib.rs

```diff
@@ -97,116 +98,175 @@ fn init() {
         .with(tracing_subscriber::EnvFilter::from_default_env())
         .init();
 
-    debug!("init called");
+    debug!("Initializing mirrord-layer.");
+
     let config = Config::init_from_env().unwrap();
+
     let pf = RUNTIME.block_on(pod_api::create_agent(
         &config.impersonated_pod_name,
         &config.impersonated_pod_namespace,
         &config.agent_namespace,
         config.agent_rust_log,
         config.agent_image,
     ));
+
     let (sender, receiver) = channel::<HookMessage>(1000);
     unsafe {
         HOOK_SENDER = Some(sender);
     };
+
     enable_hooks();
+
     RUNTIME.spawn(poll_agent(pf, receiver));
 }
 
+async fn handle_hook_message(
+    codec: &mut actix_codec::Framed<impl AsyncRead + AsyncWrite + Unpin, ClientCodec>,
+    open_file_handler: &Mutex<Vec<oneshot::Sender<mirrord_protocol::FileOpenResponse>>>,
+    port_mapping: &mut HashMap<Port, ListenData>,
+    hook_message: Option<HookMessage>,
+) {
+    match hook_message {
+        Some(HookMessage::Listen(listen)) => {
+            debug!("HookMessage::Listen {listen:#?}");
+
+            codec
+                .send(ClientMessage::PortSubscribe(vec![listen.real_port]))
+                .await
+                .unwrap();
+
+            port_mapping.insert(
+                listen.real_port,
+                ListenData {
+                    port: listen.fake_port,
+                    ipv6: listen.ipv6,
+                    fd: listen.fd,
+                },
+            );
+        }
+        Some(HookMessage::OpenFileHook(open)) => {
+            debug!("HookMessage::OpenFileHook {open:#?}");
+
+            // NOTE(alex): Lock the file handler and insert a channel that will be used to retrieve
+            // the file data when it comes back from `DaemonMessage::OpenFileResponse`.
+            open_file_handler.lock().unwrap().push(open.file_channel_tx);
```

Comment:
I wouldn't choose a vec mechanism as this won't work if we get two open at the same time (threads), it's kinda error prune. I'd just use a hashmap of request id to response id?